### PR TITLE
Installer rewrite w/ category support

### PR DESCRIPTION
**Pull request by Kartoffel**
Major installer rewrite, adds category support and has a clearer indication of button functions.

[Demo](https://www.youtube.com/watch?v=wpr_rIzz6Rg)

### CODE REVIEW

**Comment by Roosted7** [comment]
Like it :+1:

File: esp32/modules/installer.py [hunk 1]
```diff
@@ -16,87 +16,163 @@
...
-ugfx.input_attach(ugfx.BTN_A, woezel_it)
-ugfx.input_attach(ugfx.BTN_B, start_launcher)
-
-ugfx.input_attach(ugfx.BTN_START, start_app)
+options = ugfx.List(0,0,int(ugfx.width()/2),ugfx.height())
 
 text = ugfx.Textbox(int(ugfx.width()/2),0, int(ugfx.width()/2), ugfx.height())
+text.text("Downloading category list...")
 
 ugfx.set_lut(ugfx.LUT_FULL)
```
Comment (Roosted7) [inline esp32/modules/installer.py]: You can pass on the LUT to .flush()

File: esp32/modules/installer.py [hunk 2]
```diff
@@ -16,87 +16,163 @@
...
+	ugfx.input_attach(ugfx.JOY_DOWN, 0)
+	ugfx.input_attach(ugfx.BTN_A, 0)
+	ugfx.input_attach(ugfx.BTN_B, 0)
+	ugfx.input_attach(ugfx.BTN_START, 0)
+
+	empty_options()
+	text.destroy()
+	text = ugfx.Textbox(int(ugfx.width()/2),0, int(ugfx.width()/2), ugfx.height())
+	text.text("Downloading list of eggs...")
+	ugfx.set_lut(ugfx.LUT_FULL)
```
Comment (Roosted7) [inline esp32/modules/installer.py]: And here

**Review by Roosted7 — APPROVED** [review]
Looks good

**Comment by Kartoffel** [comment]
Changed the LUT thing, please check @Roosted7

**Review by annejan — APPROVED** [review]
Awesome . .

**Pull request merged**